edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/ScopeIdentitiesHandler.cs

```diff
@@ -20,97 +13,41 @@ public class ScopeIdentitiesHandler : IMessageProducer
     {
         const string Topic = "$internal/identities";
         readonly Task<IDeviceScopeIdentitiesCache> deviceScopeIdentitiesCacheGetter;
-        readonly AsyncLock cacheUpdate = new AsyncLock();
-        IMqttBrokerConnector connector;
+        readonly NotificationHandler<IList<string>> notificationHandler;
 
         public ScopeIdentitiesHandler(Task<IDeviceScopeIdentitiesCache> deviceScopeIdentitiesCacheGetter)
         {
             this.deviceScopeIdentitiesCacheGetter = deviceScopeIdentitiesCacheGetter;
+            this.notificationHandler = new NotificationHandler<IList<string>>(this.ConvertNotificationToMessagesAsync, storedNotificationRetriever: this.RetrieveDeviceScopeIdentitiesCacheAsync);
         }
 
-        public void SetConnector(IMqttBrokerConnector connector)
-        {
-            this.connector = connector;
-            this.connector.OnConnected += async (sender, args) => await this.OnConnect();
-        }
+        public void SetConnector(IMqttBrokerConnector connector) => this.notificationHandler.SetConnector(connector);
 
-        async Task OnConnect()
+        async Task<IEnumerable<Message>> RetrieveDeviceScopeIdentitiesCacheAsync()
         {
-            using (await this.cacheUpdate.LockAsync())
-            {
-                var deviceScopeIdentitiesCache = await this.deviceScopeIdentitiesCacheGetter;
-                deviceScopeIdentitiesCache.ServiceIdentitiesUpdated +=
-                    async (sender, serviceIdentities) => await this.ServiceIdentitiesUpdated(serviceIdentities);
+            var deviceScopeIdentitiesCache = await this.deviceScopeIdentitiesCacheGetter;
+            deviceScopeIdentitiesCache.ServiceIdentitiesUpdated += async (sender, serviceIdentities) => await this.notificationHandler.NotifyAsync(serviceIdentities);
```

Comment:
>ServiceIdentitiesUpdated [](start = 39, length = 24)

ServiceIdentityRemoved should also be hooked up (I think this was missing before as well)